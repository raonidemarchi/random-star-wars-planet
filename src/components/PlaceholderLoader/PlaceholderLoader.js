import React from 'react';
import styles from './PlaceholderLoader.module.scss';

function PlaceholderLoader({ height = '14px', maxWidth = '100%', marginBottom = '0' }) {
  return (
    <div className={styles.placeholder} style={{ minHeight: height, maxWidth, marginBottom }}>
      <div className={styles.animatedBackground} style={{ height }}></div>
    </div>
  );
}

export default PlaceholderLoader;
