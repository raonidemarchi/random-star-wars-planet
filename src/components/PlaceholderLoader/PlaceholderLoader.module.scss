.placeholder {
  margin: 0 auto;
  max-width: 90%;
  border-radius: 2px;
  background-color: #616161;
  
  .animatedBackground {
    animation-duration: 1.25s;
    animation-fill-mode: forwards;
    animation-iteration-count: infinite;
    animation-name: placeHolderShimmer;
    animation-timing-function: linear;
    background: darkgray;
    background: linear-gradient(to right, #616161 10%, #757575 18%, #616161 33%);
    background-size: 800px 104px;
    border-radius: 2px;
    position: relative;
  }

  @keyframes placeHolderShimmer {
    0% {
      background-position: -468px 0
    }
    100% {
      background-position: 468px 0
    }
  }
}