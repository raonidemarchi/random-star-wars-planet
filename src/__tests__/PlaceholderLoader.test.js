import React from 'react';
import PlaceholderLoader from '../components/PlaceholderLoader/PlaceholderLoader';
import TestRenderer from 'react-test-renderer';

describe('PlaceholderLoader component', () => {
  test('Should render a PlaceholderLoader', () => {
    const tree = TestRenderer.create(
      <PlaceholderLoader />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});