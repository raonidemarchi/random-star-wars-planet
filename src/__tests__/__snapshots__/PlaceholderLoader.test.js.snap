// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceholderLoader component Should render a PlaceholderLoader 1`] = `
<div
  className="placeholder"
  style={
    Object {
      "marginBottom": "0",
      "maxWidth": "100%",
      "minHeight": "14px",
    }
  }
>
  <div
    className="animatedBackground"
    style={
      Object {
        "height": "14px",
      }
    }
  />
</div>
`;
